src/SpacetimeDBClient.cs

```diff
@@ -201,122 +204,138 @@ HashSet<byte[]> GetInsertHashSet(System.Type tableType, int tableSize)
                                 continue;
                             }
 
-                            var hashSet = GetInsertHashSet(table.ClientTableType, subscriptionUpdate.TableUpdates.Count);
-
-                            foreach (var row in update.TableRowOperations)
+                            if (update.Deletes.Count != 0)
                             {
-                                var rowBytes = row.Row.ToByteArray();
+                                Logger.LogWarning("Non-insert during a subscription update!");
+                            }
 
-                                if (row.Op != TableRowOperation.Types.OperationType.Insert)
-                                {
-                                    Logger.LogWarning("Non-insert during a subscription update!");
-                                    continue;
-                                }
+                            var hashSet = GetInsertHashSet(table.ClientTableType, initialSubscription.DatabaseUpdate.Tables.Count);
 
-                                if (!hashSet.Add(rowBytes))
-                                {
-                                    // Ignore duplicate inserts in the same subscription update.
-                                    continue;
-                                }
+                            foreach (var row in update.Inserts)
+                            {
+                                switch (row) {
+                                    case EncodedValue.Binary(var bin):
+                                        if (!hashSet.Add(bin)) {
+                                            // Ignore duplicate inserts in the same subscription update.
+                                            continue;
+                                        }
 
+                                        var obj = table.DecodeValue(bin);
+                                        var op = new DbOp {
+                                            table = table,
+                                            insert = new(obj, bin),
+                                        };
 
-                                var obj = table.DecodeValue(row.Row);
-                                var op = new DbOp
-                                {
-                                    table = table,
-                                    insert = new(obj, rowBytes),
-                                };
+                                        dbOps.Add(op);
+                                        break;
 
-                                dbOps.Add(op);
+                                    case EncodedValue.Text(var txt):
+                                        throw new InvalidOperationException("TODO?");
```

Comment:
In this case let's just print a warning and skip the row instead of throwing an exception.

Reply:
Good point.

And besides, the SDK should only communicate in BSATN with the server now.